packages/anoncreds-rs/src/services/AnonCredsRsHolderService.ts

```diff
@@ -0,0 +1,380 @@
+        schemas: rsSchemas,
+        presentationRequest: PresentationRequest.load(JSON.stringify(proofRequest)),
+        credentials: credentials.map((entry) => entry.credentialEntry),
+        credentialsProve,
+        selfAttest: requestedCredentials.selfAttestedAttributes,
+        masterSecret: MasterSecret.load(JSON.stringify({ value: { ms: linkSecretRecord.value } })),
+      })
+
+      return JSON.parse(presentation.toJson())
+    } catch (error) {
+      agentContext.config.logger.error(`Error creating AnonCreds Proof`, {
+        error,
+        proofRequest,
+        requestedCredentials,
+      })
+      throw new AnonCredsRsError(`Error creating proof: ${error}`, { cause: error })
+    }
+  }
+
+  public async createCredentialRequest(
+    agentContext: AgentContext,
+    options: CreateCredentialRequestOptions
+  ): Promise<CreateCredentialRequestReturn> {
+    const { credentialDefinition, credentialOffer } = options
+    try {
+      const linkSecretRepository = agentContext.dependencyManager.resolve(AnonCredsLinkSecretRepository)
+
+      // If a link secret is specified, use it. Otherwise, attempt to use default link secret
+      const linkSecretRecord = options.linkSecretId
+        ? await linkSecretRepository.getByLinkSecretId(agentContext, options.linkSecretId)
+        : await linkSecretRepository.findDefault(agentContext)
+
+      if (!linkSecretRecord) {
+        // No default link secret: TODO: shall we create a new one?
+        throw new AnonCredsRsError('Link Secret not found')
+      }
+
+      const { credentialRequest, credentialRequestMetadata } = CredentialRequest.create({
+        credentialDefinition: CredentialDefinition.load(JSON.stringify(credentialDefinition)),
+        credentialOffer: CredentialOffer.load(JSON.stringify(credentialOffer)),
+        masterSecret: MasterSecret.load(JSON.stringify({ value: { ms: linkSecretRecord.value } })),
+        masterSecretId: linkSecretRecord.linkSecretId,
+      })
+
+      return {
+        credentialRequest: JSON.parse(credentialRequest.toJson()),
+        credentialRequestMetadata: JSON.parse(credentialRequestMetadata.toJson()),
+      }
+    } catch (error) {
+      throw new AnonCredsRsError(`Error creating credential request: ${error}`, { cause: error })
+    }
+  }
+
+  public async storeCredential(agentContext: AgentContext, options: StoreCredentialOptions): Promise<string> {
+    const linkSecretRecord = await agentContext.dependencyManager
+      .resolve(AnonCredsLinkSecretRepository)
+      .getByLinkSecretId(agentContext, options.credentialRequestMetadata.master_secret_name)
+
+    // TODO: In order to support all tags from AnonCreds spec (section 9.1.1 about Restrictions), we need to tag
+    // the credential records with some properties of the schema it is based on. This means that Schema should have been
+    // previously retrieved and present in the storage. Can we assume that at this point it will be stored?
```

Comment:
I don't think we store it when we receive a credential right? So in this case we should require the `AnonCredsSchema` as input to this method  I think? Same as we require the `CredentialDefinition`

Reply:
I wrote this comment some days ago and today, when writing the E2E issuance demo, I figured out it is not stored (neither the CredentialDefinition), so I had to add some lines to manually save an AnonCredsSchemaRecord to make it work.

I think the best would be to require the AnonCredsSchema and leave the CredentialFormatService the task of retrieving it.